controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/StreamRequestHandler.java

```diff
@@ -100,7 +100,10 @@ public StreamRequestHandler(AutoScaleTask autoScaleTask,
                                                                                T event,
                                                                                Predicate<Throwable> writeBackPredicate) {
         CompletableFuture<Void> result = new CompletableFuture<>();
-        task.execute(event)
+        Retry.withExpBackoff(100, 10, 6, 10000)
```

Comment:
This retry block is added for 'TaskExceptions.StartException'.
The model for processing is as follows: 
For any user initiated workflow, the caller's invoked method posts an event in the requeststream and then performs first step of the workflow. 

If the event is picked before caller has called "first" step then its processing is supposed to throw TaskStartException. 
StreamRequestHandler, upon receiving the taskStartException effectively waits and retries several times (giving the caller's method ample time to start the processing) before it discards the event for processing. 

The idea is as follows: if we were to repost this event, then it can linger in the stream and have duplicates reposted ad infinitum in case caller failed to start the work.

A caller can fail to start the work for various reasons ->
1. after posting the event, the controller processing the caller's request may crash.
2. the request may not be allowed hence the call failed.
The contract we have with the caller is that if we are able to start the work (by performing the first step) then we respond to them with acknowledgement that their request is accepted. 
So we want to perform event processing iff caller's method was able to perform the "first step". 
if the caller has not started the work while our retries exhaust, it wouldnt have started the work.